main/main.cpp

```diff
@@ -1699,8 +1699,23 @@ bool Main::start() {
 			return false;
 		}
 
-	} else {
-		main_loop_type = GLOBAL_DEF("application/run/main_loop_type", "");
+	} else { // Not based on script path.
+		if (!editor && !ClassDB::class_exists(main_loop_type) && ScriptServer::is_global_class(main_loop_type)) {
+			String script_path = ScriptServer::get_global_class_path(main_loop_type);
+			Ref<Script> script_res = ResourceLoader::load(script_path, "Script", true);
```

Comment:
```
- Ref<Script> script_res = ResourceLoader::load(script_path);
+ Ref<Script> script_res = ResourceLoader::load(script_path, "Script", true);
```

script_res should be load without any cache

Reply:
I think this shouldn't matter much because not a lot of resources get loaded at engine startup to worry about this I guess.

Reply:
most of the time, the loading of custom main loop is fine.

some times. if I clike Run Button very fast after opening the editor. then my custom main loop script seems load failed. 

![QQ20210907001802](https://user-images.githubusercontent.com/6326007/132244847-443cdf71-370a-483a-84dd-fd7973eed190.png)

After this error happening, script wont be reload until I reopen the editor. 

tested by setting no_cache=true here still won't fix this problem.

Reply:
I think this feature may just reveal editor problem. According to error, the base type is basically `""`, which probably means that editor settings could not get loaded (maybe when the file gets locked for reading, and cannot be read again until the previous Godot process shuts down completely).